src/pymor/reductors/h2.py

```diff
@@ -555,13 +555,15 @@ class TFIRKAReductor(GenericIRKAReductor):
     Parameters
     ----------
     fom
-        The full-order |Model| with `eval_tf` and `eval_dtf` methods.
+        The full-order |Model| with `eval_tf` and `eval_dtf` methods that
+        should be defined at least over the open right half of the complex
+        plane.
     mu
         |Parameter values|.
     """
 
     def __init__(self, fom, mu=None):
-        assert isinstance(fom, InputOutputModel)
+        assert hasattr(fom, 'eval_tf') and hasattr(fom, 'eval_dtf')
```

Comment:
I would just let the reductor work on `TransferFunctions`. As a convenience, `__init__` could take a `Model` and use its `transfer_function`.

Reply:
Aha, yes, that's a good point.